pkg/authorization/api/conversion.go

```diff
@@ -0,0 +1,199 @@
+			NonResourceURLs: rule.NonResourceURLs.List(),
+		}
+		rules = append(rules, r)
+	}
+	return rules
+}
+
+func convertOriginSubjects(in []api.ObjectReference) ([]rbac.Subject, error) {
+	subjects := make([]rbac.Subject, 0, len(in))
+	for _, subject := range in {
+		s := rbac.Subject{
+			Name:       subject.Name,
+			APIVersion: rbac.GroupName,
+		}
+
+		switch subject.Kind {
+		case ServiceAccountKind:
+			s.Kind = rbac.ServiceAccountKind
+			s.Namespace = subject.Namespace
+		case UserKind, SystemUserKind:
+			s.Kind = rbac.UserKind
+		case GroupKind, SystemGroupKind:
+			s.Kind = rbac.GroupKind
+		default:
+			return nil, fmt.Errorf("invalid kind for origin subject: %q", subject.Kind)
+		}
+
+		subjects = append(subjects, s)
+	}
+	return subjects, nil
+}
+
+func convertOriginRoleRef(in *api.ObjectReference) rbac.RoleRef {
+	return rbac.RoleRef{
+		APIGroup: rbac.GroupName,
+		Kind:     getKind(in.Namespace),
+		Name:     in.Name,
+	}
+}
+
+func getKind(namespace string) string {
+	kind := "ClusterRole"
+	if len(namespace) != 0 {
+		kind = "Role"
+	}
+	return kind
+}
+
+func Convert_rbac_ClusterRole_To_api_ClusterRole(in *rbac.ClusterRole, out *ClusterRole, _ conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Rules = convertRBACPolicyRules(in.Rules)
+	return nil
+}
+
+func Convert_rbac_Role_To_api_Role(in *rbac.Role, out *Role, _ conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Rules = convertRBACPolicyRules(in.Rules)
+	return nil
+}
+
+func Convert_rbac_ClusterRoleBinding_To_api_ClusterRoleBinding(in *rbac.ClusterRoleBinding, out *ClusterRoleBinding, _ conversion.Scope) error {
+	var err error
+	if out.Subjects, err = convertRBACSubjects(in.Subjects); err != nil {
+		return err
+	}
+	out.RoleRef = convertRBACRoleRef(&in.RoleRef, "")
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func Convert_rbac_RoleBinding_To_api_RoleBinding(in *rbac.RoleBinding, out *RoleBinding, _ conversion.Scope) error {
+	var err error
+	if out.Subjects, err = convertRBACSubjects(in.Subjects); err != nil {
+		return err
+	}
+	out.RoleRef = convertRBACRoleRef(&in.RoleRef, in.Namespace)
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func convertRBACSubjects(in []rbac.Subject) ([]api.ObjectReference, error) {
+	subjects := make([]api.ObjectReference, 0, len(in))
+	for _, subject := range in {
+		s := api.ObjectReference{
+			Name: subject.Name,
+		}
+
+		switch subject.Kind {
+		case rbac.ServiceAccountKind:
+			s.Kind = ServiceAccountKind
+			s.Namespace = subject.Namespace
+		case rbac.UserKind:
+			s.Kind = determineUserKind(subject.Name, validation.ValidateUserName)
+		case rbac.GroupKind:
+			s.Kind = determineGroupKind(subject.Name, validation.ValidateGroupName)
+		default:
+			return nil, fmt.Errorf("invalid kind for rbac subject: %q", subject.Kind)
+		}
+
+		subjects = append(subjects, s)
+	}
+	return subjects, nil
+}
+
+func convertRBACRoleRef(in *rbac.RoleRef, namespace string) api.ObjectReference {
```

Comment:
Can you name these functions as Convert_foo_to_bar?  Obviously the signature won't match, but it will help remind me as I read it.  Also, godoc on for the args.  `namespace` isn't obvious.